File: php/pe/Pe3.php
<?php
/**
 * Solve project euler 3
 *
 * Find the largest prime factor of a composite number.
 */

namespace Pe;

class Pe3
{
    /**
     * @param int $n
     * @return int[]
     */
    public static function calc(int $n): array
    {
        return get_divisors($n);
    }
}

File: rs/pe/src/pe3.rs
/// # Solve project euler 3
///
/// Find the largest prime factor of a composite number.
use utils::get_divisors;

pub fn calc(n: u64) -> Vec<u64> {
    return get_divisors(n);
}

pub fn pe3() {
    let s: Vec<u64> = calc(600851475143);
    println!("{:?}", s);
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test() {
        let ps = calc(600851475143);
        assert_eq!(ps, [71, 839, 1471, 6857]);
    }
}
